x-pack/plugins/apm/public/components/app/settings/agent_explorer/agent_instances/agent_contextual_information/index.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { TypeOf } from '@kbn/typed-react-router-config';
+import React from 'react';
+import { AgentExplorerFieldName } from '../../../../../../../common/agent_explorer';
+import { NOT_AVAILABLE_LABEL } from '../../../../../../../common/i18n';
+import { AgentName } from '../../../../../../../typings/es_schemas/ui/fields/agent';
+import { ApmRoutes } from '../../../../../routing/apm_route_config';
+import { ServiceLink } from '../../../../../shared/service_link';
+import { StickyProperties } from '../../../../../shared/sticky_properties';
+import { TruncateWithTooltip } from '../../../../../shared/truncate_with_tooltip';
+import { AgentExplorerDocsLink } from '../../agent_explorer_docs_link';
+
+function formatString(value?: string | null) {
+  return value || NOT_AVAILABLE_LABEL;
+}
+
+export function AgentContextualInformation({
+  agentName,
+  serviceName,
+  agentDocsPageUrl,
+  instances,
+  query,
+}: {
+  agentName?: AgentName;
+  serviceName?: string;
+  agentDocsPageUrl?: string;
+  instances?: number;
```

Comment:
Why are these props optional? I assume this component is only shown when there is an agent available?